src/charm.py

```diff
@@ -5,49 +5,427 @@
+
+    def _gen_pebble_plan(self) -> ops.pebble.LayerDict:
+        """Generate penpot pebble plan.
+
+        Returns:
+            Penpot pebble plan.
+        """
+        plan = ops.pebble.LayerDict(
+            summary="penpot services",
+            description="penpot services",
+            services={
+                "frontend": {
+                    "command": './nginx-entrypoint.sh nginx -g "daemon off;"',
+                    "working-dir": "/opt/penpot/frontend/",
+                    "override": "replace",
+                    "after": ["backend"],
+                    "environment": {
+                        "PENPOT_BACKEND_URI": "http://127.0.0.1:6060",
+                        "PENPOT_EXPORTER_URI": self._get_penpot_exporter_uri(),
+                        "PENPOT_INTERNAL_RESOLVER": self._get_local_resolver(),
+                        "PENPOT_FLAGS": " ".join(self._get_penpot_frontend_options()),
+                    },
+                },
+                "backend": {
+                    "command": "/opt/penpot/backend/run.sh",
+                    "override": "replace",
+                    "working-dir": "/opt/penpot/backend/",
+                    "environment": {
+                        "JAVA_HOME": "/usr/lib/jvm/java-21-openjdk-amd64",
+                        "PENPOT_TELEMETRY_ENABLED": "false",
+                        "PENPOT_PUBLIC_URI": typing.cast(str, self._get_public_uri()),
+                        "PENPOT_FLAGS": " ".join(self._get_penpot_backend_options()),
+                        **typing.cast(dict[str, str], self._get_penpot_secret_key()),
+                        **typing.cast(dict[str, str], self._get_postgresql_credentials()),
+                        **typing.cast(dict[str, str], self._get_redis_credentials()),
+                        **typing.cast(dict[str, str], self._get_smtp_credentials()),
+                        **typing.cast(dict[str, str], self._get_s3_credentials()),
+                    },
+                },
+                "exporter": {
+                    "command": "node app.js",
+                    "working-dir": "/opt/penpot/exporter/",
+                    "override": "replace",
+                    "after": ["backend", "frontend"],
+                    "environment": {
+                        "PENPOT_PUBLIC_URI": "http://127.0.0.1:8080",
+                        "PLAYWRIGHT_BROWSERS_PATH": "/opt/penpot/exporter/browsers",
+                        **typing.cast(dict[str, str], self._get_redis_credentials()),
+                    },
+                },
+            },
+            checks={
+                "backend-ready": {
+                    "override": "replace",
+                    "level": "ready",
+                    "period": "30s",
+                    "exec": {
+                        # pylint: disable=line-too-long
+                        "command": 'bash -c "pebble services backend | grep -q inactive || curl -f localhost:6060/readyz"'  # noqa: E501
```

Comment:
Adding a timeout to curl might help not get stuck (just in case).

Reply:
Yes, I think that's a good idea, updated, thanks!